server/src/client_service.rs

```diff
@@ -566,84 +580,135 @@ mod tests {
-        assert_eq!(resp.get_quote_added(), added_quote);
+        // We should see this sci if we call get quotes
+        let mut req = GetQuotesRequest::default();
+        req.set_pair((&pair).into());
+        let resp = client_api.get_quotes(&req).expect("get quotes failed");
+        let received_scis = resp
+            .get_quotes()
+            .iter()
+            .map(|quote| SignedContingentInput::try_from(quote.get_sci()).unwrap())
+            .collect::<Vec<_>>();
+        assert_eq!(received_scis, vec![sci.clone()]);
+
+        // Build a Tx and add its outputs, and the sci's key image, to the ledger
+        let block_version = BlockVersion::MAX;
+        let fog_resolver = MockFogResolver::default();
+
+        let offerer_account = AccountKey::random(&mut rng);
+
+        let tx = get_transaction(
+            block_version,
+            TokenId::from(0),
+            2,
+            2,
+            &offerer_account,
+            &offerer_account,
+            fog_resolver,
+            &mut rng,
+        )
+        .unwrap();
+        add_txos_and_key_images_to_ledger(
+            &mut ledger,
+            BlockVersion::MAX,
+            tx.prefix.outputs,
+            vec![key_image],
+            &mut rng,
+        )
+        .unwrap();
 
-        // Remove the quote, we should see a live update.
-        let mut req = RemoveQuoteRequest::default();
-        req.set_quote_id(added_quote.get_id().clone());
+        assert_eq!(ledger.num_blocks().unwrap(), starting_blocks + 1);
 
-        let _resp = client_api.remove_quote(&req).expect("remove quote failed");
+        // Block until we have processed this block
+        let mut tries = 500usize;
+        while quote_book.get_current_block_index() + 1 < ledger.num_blocks().unwrap() {
+            std::thread::sleep(std::time::Duration::from_millis(100));
+            tries -= 1;
+            assert_ne!(tries, 0, "Quote book failed to catch up in time");
+        }
 
-        let resp = rx.recv().expect("recv failed");
-        assert_eq!(resp.get_quote_removed(), added_quote.get_id());
+        // We should see an empty book now if we call get quotes
+        let mut req = GetQuotesRequest::default();
+        req.set_pair((&pair).into());
+        let resp = client_api.get_quotes(&req).expect("get quotes failed");
+        let received_scis = resp
+            .get_quotes()
+            .iter()
+            .map(|quote| SignedContingentInput::try_from(quote.get_sci()).unwrap())
+            .collect::<Vec<_>>();
+        assert_eq!(received_scis, vec![]);
+
+        // If we try to add the same quote back, it should be rejected as stale
+        let req = SubmitQuotesRequest {
+            quotes: vec![(&sci).into()].into(),
+            ..Default::default()
+        };
+        let resp = client_api.submit_quotes(&req).expect("submit quote failed");
+        assert_eq!(resp.status_codes, vec![QuoteStatusCode::QUOTE_IS_STALE]);
```

Comment:
i added this test so that we can verify this, but it's no longer doing the funky remove_quote_callback

Reply:
I guess it's still possible to do this test without using the ledger, if we use like a mock synchronized ledger or something. but i feel like it's easier to just use a ledger and a real synchronized ledger